### PR TITLE
LGA-2803 - Add maintenance page

### PR DESCRIPTION
**Pull request by said-moj**
## What does this pull request do?

Add maintenance page

## Any other changes that would benefit highlighting?

Intentionally left blank.

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: app/templates/main/503.html [hunk 1]
```diff
@@ -1,14 +1,34 @@
 {% extends "base.html" %}
 
-{% block pageTitle %}Sorry, the service is unavailable – {{config['SERVICE_NAME']}} – GOV.UK{% endblock %}
+{% block pageTitle %}Sorry, the service is unavailable{% endblock %}
```
Comment (BenMillar-MOJ) [inline app/templates/main/503.html:3]: According to [https://design-system.service.gov.uk/patterns/service-unavailable-pages/](https://design-system.service.gov.uk/patterns/service-unavailable-pages/).
The page title should be ‘Sorry, the service is unavailable – service name – GOV.UK’

File: helm_deploy/laa-access-civil-legal-aid/templates/ingress.yaml [hunk 2]
```diff
@@ -13,6 +13,7 @@ metadata:
   labels:
     {{- include "laa-access-civil-legal-aid.labels" . | nindent 4 }}
   annotations:
+    nginx.ingress.kubernetes.io/custom-http-errors: "413,502,504"
```
Comment (BenMillar-MOJ) [inline helm_deploy/laa-access-civil-legal-aid/templates/ingress.yaml:16]: Can you clarify my understanding of this please.
Does this forward 413's, 502's and 504's to Cloud Platform's default Ngninx error page?

File: app/main/routes.py [hunk 3]
```diff
@@ -100,3 +113,17 @@ def http_exception(error):
 def csrf_error(error):
     flash("The form you were submitting has expired. Please try again.")
     return redirect(request.full_path)
+
+
+@bp.before_request
```
Comment (BenMillar-MOJ) [inline app/main/routes.py:118]: How does this handle requests going to other blueprints?

**Review by TawneeOwl — APPROVED** [review]
LGTM

File: app/main/routes.py (hunk 3 above)
Comment (said-moj) [reply app/main/routes.py:118]: Decided to use nested blue prints, so this should not be a problem

File: app/templates/main/503.html (hunk 1 above)
Comment (said-moj) [reply app/templates/main/503.html]: Fixed, thanks

File: app/main/routes.py [hunk 4]
```diff
@@ -100,3 +113,17 @@ def http_exception(error):
...
+def service_unavailable_middleware():
+    if not current_app.config["SERVICE_UNAVAILABLE"]:
+        return
+
+    service_unavailable_url = url_for("main.service_unavailable_page")
+    exempt_urls = [
+        service_unavailable_url,
+        url_for("main.status"),
```
Comment (BenMillar-MOJ) [inline app/main/routes.py:126]: I believe the `locale` endpoint should be appended to this list. Currently the language switcher doesn't work as expected when on the maintenance page.

Comment (BenMillar-MOJ) [reply app/main/routes.py:126]: Additionally, the following pages may need to be added to this list:

- [Accessibility](https://access-civil-legal-aid-uat.cloud-platform.service.justice.gov.uk/accessibility) 
- [Cookies](https://access-civil-legal-aid-uat.cloud-platform.service.justice.gov.uk/cookies) 
- [Privacy](https://access-civil-legal-aid-uat.cloud-platform.service.justice.gov.uk/privacy)

As these pages are linked to from the maintenance page.

**Review by BenMillar-MOJ — CHANGES_REQUESTED** [review]

File: app/main/routes.py (hunk 4 above)
Comment (said-moj) [reply app/main/routes.py:126]: Good spot, this has been updated now and all those footer links should work

**Review by BenMillar-MOJ — APPROVED** [review]
LGTM, thank you for this! 👍

**Comment by BenMillar-MOJ** [comment]
I raised a PR to use `app_context_processor` rather than a BP specific processor, this should fix the issue we discussed earlier [https://github.com/ministryofjustice/laa-access-civil-legal-aid/pull/44/files](https://github.com/ministryofjustice/laa-access-civil-legal-aid/pull/44/files)

**Pull request merged**